ios/CodePush/CodePush.m

```diff
@@ -783,19 +783,35 @@ - (void)applicationWillResignActive
         NSString *appVersion = [[CodePushConfig current] appVersion];
         resolve([CodePushTelemetryManager getBinaryUpdateReport:appVersion]);
         return;
+    } else {
+        NSDictionary *retryStatusReport = [CodePushTelemetryManager getRetryStatusReport];
+        if (retryStatusReport) {
+            resolve(retryStatusReport);
+            return;
+        }
     }
 
     resolve(nil);
 }
 
+RCT_EXPORT_METHOD(recordStatusReported:(NSDictionary *)statusReport)
+{
+    [CodePushTelemetryManager recordStatusReported:statusReport];
+}
+
+RCT_EXPORT_METHOD(saveStatusReportForRetry:(NSDictionary *)statusReport)
```

Comment:
Do you have a better suggestion? I considered "queue" instead of "save" but it's not really putting anything in a queue..